tests/framework/results/test_context.py

```diff
@@ -456,12 +482,13 @@ def test_gather_results_partial_stratifications_in_results(
     for results, _measure, _formatter in ctx.gather_results(
         population, lifecycle_phase, mocked_event
     ):
-        unladen_results = results.reset_index().query('familiar=="unladen_swallow"')
-        assert len(unladen_results) > 0
-        assert (unladen_results[VALUE_COLUMN] == 0).all()
+        if results is not None:
```

Comment:
You should `assert results is not None`. Otherwise we won't detect if a bug is introduced that makes `result` None